Sources/ArcGISToolkit/Components/Offline/Utilities/FileManager+Offline.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Foundation
+
+extension FileManager {
+    /// The path to the documents folder.
+    private var documentsDirectory: URL {
+        URL.documentsDirectory
+    }
+    
+    /// The path to the offline map areas directory within the documents directory.
+    /// `Documents/OfflineMapAreas/`
+    private var offlineMapAreasDirectory: URL {
+        documentsDirectory.appending(component: "OfflineMapAreas/")
+    }
+    
+    /// The path to the web map directory for a specific portal item.
+    /// `Documents/OfflineMapAreas/<Portal Item ID>/`
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    private func portalItemDirectory(forPortalItemID portalItemID: PortalItem.ID) -> URL {
```

Comment:
This should be on `URL`, not `FileManager`. Please move this to a private URL extension below this extension.

Reply:
I tried to move these methods to a URL extension but `preplannedDirectory(forPortalItemID:preplannedMapAreaID)` needs to be accessible outside this file and for some reason the method is only available as `URL.preplannedDirectory(self: URL)`.

Reply:
Try `static func preplannedDirectory`

Reply:
Nevermind, I just applied this feedback.